### PR TITLE
Add TPU v4 blueprint and tutorial to demonstrate running TPU workload

### PR DESCRIPTION
**Pull request by harshthakkar01**
This PR runs TPU benchmark test `maxtest` using Slurm Cluster + TPU v4. This also contains tutorial document to demonstrate how to run TPU workload using this blueprint. 

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: community/examples/hpc-slurm6-tpu.yaml [hunk 1]
```diff
@@ -19,22 +19,50 @@ blueprint_name: slurm6-tpu
...
+        content: |
+          #!/bin/bash
+
+          virtualenv venv
+          source venv/bin/activate
+
+          git clone https://github.com/google/maxtext
+          cd maxtext/
+          bash setup.sh
+          python3 MaxText/train.py MaxText/configs/base.yml run_name=1xv4-128 base_output_directory=/opt/apps/scripts/tpu-test/output/ dataset_path=gs://dataset_tpu/dataset async_checkpointing=False attention=dot_product steps=100
```
Comment (harshthakkar01) [inline community/examples/hpc-slurm6-tpu.yaml:56]: TODO: remove this

Comment (harshthakkar01) [reply community/examples/hpc-slurm6-tpu.yaml]: Outdated.

File: community/examples/hpc-slurm6-tpu-v4.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+blueprint_name: slurm6-tpu
+
+vars:
+  project_id:
```
Comment (mr0re1) [inline community/examples/hpc-slurm6-tpu-v4.yaml:20]: nit. Please add `## Set GCP Project ID Here ##`

File: community/examples/hpc-slurm6-tpu-v4.yaml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  modules:
+  - id: script
+    source: modules/scripts/startup-script
+    settings:
+      runners:
+      - type: shell
+        destination: tpu_setup.sh
+        content: |
+          #!/bin/bash
+
```
Comment (mr0re1) [inline community/examples/hpc-slurm6-tpu-v4.yaml:41]: nit. remove newline

**Review by mr0re1 — APPROVED** [review]

File: community/examples/hpc-slurm6-tpu-v4.yaml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+blueprint_name: slurm6-tpu
+
+vars:
+  project_id:
+  deployment_name: slurm6-tpu
+  region: us-central2
+  zone: us-central2-b
+
+deployment_groups:
+- group: primary
```
Comment (mr0re1) [inline community/examples/hpc-slurm6-tpu-v4.yaml:26]: Don't use separate group for VPC

Comment (harshthakkar01) [reply community/examples/hpc-slurm6-tpu-v4.yaml:26]: Removed.

File: community/examples/hpc-slurm6-tpu-v4.yaml (hunk 3 above)
Comment (harshthakkar01) [reply community/examples/hpc-slurm6-tpu-v4.yaml]: Removed.

**Review by mr0re1 — APPROVED** [review]

**Pull request merged**